src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -32,6 +48,221 @@
+
+
+class SyncExecutor:
+    """
+    Synchronous executor for generating outputs from inputs using a given generation function.
+
+    Args:
+        generation_fn (Callable[[Any], Any]): The generation function that takes an input and
+        returns an output.
+
+        tqdm_bar_format (Optional[str], optional): The format string for the progress bar. Defaults
+        to None.
+
+        exit_on_error (bool, optional): Whether to exit execution on the first encountered error.
+        Defaults to True.
+
+        fallback_return_value (Union[Unset, Any], optional): The fallback return value for tasks
+        that encounter errors. Defaults to _unset.
+    """
+
+    def __init__(
+        self,
+        generation_fn: Callable[[Any], Any],
+        tqdm_bar_format: Optional[str] = None,
+        exit_on_error: bool = True,
+        fallback_return_value: Union[Unset, Any] = _unset,
+    ):
+        self.generate = generation_fn
+        self.fallback_return_value = fallback_return_value
+        self.tqdm_bar_format = tqdm_bar_format
+        self.exit_on_error = exit_on_error
+
+        self._TERMINATE = False
+
+    def _signal_handler(self, signum: int, frame: Any) -> None:
+        tqdm.write("Process was interrupted. The return value will be incomplete...")
+        self._TERMINATE = True
+
+    def run(self, inputs: Sequence[Any]) -> List[Any]:
+        signal.signal(signal.SIGINT, self._signal_handler)
+        outputs = [self.fallback_return_value] * len(inputs)
+        progress_bar = tqdm(total=len(inputs), bar_format=self.tqdm_bar_format)
+
+        for index, input in enumerate(inputs):
+            if self._TERMINATE:
+                break
+            try:
+                result = self.generate(input)
+                outputs[index] = result
+                progress_bar.update()
+            except Exception as e:
+                tqdm.write(f"Exception in worker: {e}")
+                if self.exit_on_error:
+                    break
+                else:
+                    progress_bar.update()
+        return outputs
+
+
+def get_executor_on_sync_context(
+    sync_fn: Callable[[Any], Any],
+    async_fn: Callable[[Any], Coroutine[Any, Any, Any]],
+    concurrency: int = 3,
+    tqdm_bar_format: Optional[str] = None,
+    exit_on_error: bool = True,
+    fallback_return_value: Union[Unset, Any] = _unset,
+) -> Union[AsyncExecutor, SyncExecutor]:
+    async_executor = AsyncExecutor(
+        async_fn,
+        concurrency=concurrency,
+        tqdm_bar_format=tqdm_bar_format,
+        exit_on_error=exit_on_error,
+        fallback_return_value=fallback_return_value,
+    )
+    sync_executor = SyncExecutor(
+        sync_fn,
+        tqdm_bar_format=tqdm_bar_format,
+        exit_on_error=exit_on_error,
+        fallback_return_value=fallback_return_value,
+    )
+    try:
+        asyncio.get_running_loop()
+        if getattr(asyncio, "_nest_patched", False):
+            return async_executor
```

Comment:
there's already a utility to do this, might make sense to just centralize this in utils